docs/recipes/CodeSplitting.md

```diff
@@ -0,0 +1,138 @@
+        reduce: (state, action) => {
+            // If any reducers have been removed, clean up their state first
+            if (keysToRemove.length > 0) {
+                state = { ...state as any };
+                for (let key of keysToRemove) {
+                    delete state[key];
+                }
+                keysToRemove = [];
+            }
+
+            // Delegate to the combined reducer
+            return combinedReducer(state, action);
+        },
+
+        // Adds a new reducer with the specified key
+        add: (key, reducer) => {
+            if (!key || reducers[key]) {
+                return;
+            }
+
+            // Add the reducer to the reducer mapping
+            reducers[key] = reducer;
+
+            // Generate a new combined reducer
+            combinedReducer = combineReducers(reducers);
+        },
+
+        // Removes a reducer with the specified key
+        remove: (key: string) => {
+            if (!key || !reducers[key]) {
+                return;
+            }
+
+            // Remove it from the reducer mapping
+            delete reducers[key];
+
+            // Add the key to the list of keys to clean up
+            keysToRemove.push(key);
+
+            // Generate a new combined reducer
+            combinedReducer = combineReducers(rm);
+        }
+    };
+}
+
+const staticReducers = {
+    users: usersReducer,
+    posts: postsReducer
+};
+
+export function configureStore(initialState) {
+    const reducerManager = createReducerManager(staticReducers);
+
+    // Create a store with the root reducer function being the one exposed by the manager.
+    const store = createStore(reducerManager.reduce, initialState);
+
+    // Optional: Put the reducer manager on the store so it is easily accessible
+    store.reducerManager = reducerManager;
+}
+```
+
+To add a new reducer, one can now call `store.reducerManager.add("asyncState", asyncReducer)`.
+
+To remove a reducer, one can now call `store.reducerManager.remove("asyncState")`
+
+## Libraries and Frameworks
+There are a few good libraries out there that can help you add the above functionality automatically:
+ * [`redux-dynamic-reducer`](https://github.com/ioof-holdings/redux-dynamic-reducer)
+    * This library exposes the `addReducer` function on the Redux store to accomplish the behavior we described above. It also has React bindings which make it easier to add reducers within the React component lifecycle.
+* [`redux-dynamic-modules`](https://github.com/Microsoft/redux-dynamic-modules)
+    * This library introduces the concept of a 'Redux Module', which is a bundle of Redux artifacts (reducers, middleware) that should be dynamically loaded. It also exposes a React higher-order component to load 'modules' when areas of the application come online. Additionally, it has integrations with libraries like `redux-thunk` and `redux-saga` which also help dynamically load their artifacts (thunks, sagas). 
```

Comment:
Might as well toss in a link to the list of additional libs I've seen: https://github.com/markerikson/redux-ecosystem-links/blob/master/reducers.md#dynamic-reducer-injection